website/content/en/docs/tasks/scheduling-pods.md

```diff
@@ -241,3 +241,55 @@ If instead the spread were 5, pods could be 5, 0, 0 or 3, 2, 0, or 2, 1, 2 and s
 * Karpenter is always able to improve skew by launching new nodes in the right zones. Therefore, `whenUnsatisfiable` does not change provisioning behavior.
 
 See [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/) for details.
+
+## Persistent Volume Topology
+
+Karpenter automatically detects storage scheduling requirements and includes them in node launch decisions.
+
+In the following example, the `StorageClass` defines zonal topologies for `us-west-2a` and `us-west-2b` and binding mode `WaitForFirstConsumer`.
+When the pod is created, Karpenter follows references from the `Pod` to `PersistentVolumeClaim` to `StorageClass` and identifies that this pod requires storage in `us-west-2a` and `us-west-2b`.
+It randomly selects `us-west-2a`, provisions a node in that zone, and binds the pod to the node.
+The CSI driver creates a `PersistentVolume` according to the `PersistentVolumeClaim`, which is attached during pod initialization.
+After it is bound, the `PersistentVolume` will include a node affinity rule that specifies
+
+Later on, the pod is deleted and a new pod is created that requests the same claim. This time, Karpenter identifies that a `PersistentVolume` already exists for the `PersistentVolumeClaim`, and includes its zone `us-west-2a` in the pod's scheduling requirements.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app
+spec:
+  containers: ...
+  volumes:
+    - name: storage
+      persistentVolumeClaim:
+        claimName: ebs-claim
+---
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: ebs
+provisioner: ebs.csi.aws.com
+volumeBindingMode: WaitForFirstConsumer
+allowedTopologies:
+- matchLabelExpressions:
+  - key: topology.ebs.csi.aws.com/zone
+    values: ["us-west-2a", "us-west-2b"]
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: ebs-claim
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: ebs
+  resources:
+    requests:
+      storage: 4Gi
+```
+
+Note: The EBS CSI driver uses `topology.ebs.csi.aws.com/zone` instead of the standard `topology.kubernetes.io/zone` label. Karpenter is aware of label aliasing and translates this label into `topology.kubernetes.io/zone` in memory. When configuring a `StorageClass` for the EBS CSI Driver, you must use `topology.ebs.csi.aws.com/zone`.
```

Comment:
```suggestion
{{% alert title="Note" color="primary" %}}
The EBS CSI driver uses `topology.ebs.csi.aws.com/zone` instead of the standard `topology.kubernetes.io/zone` label. Karpenter is aware of label aliasing and translates this label into `topology.kubernetes.io/zone` in memory. When configuring a `StorageClass` for the EBS CSI Driver, you must use `topology.ebs.csi.aws.com/zone`.
{{% /alert %}}
```